### ISSUE TITLE
Exception: "Cannot dynamically create an instance of type Reason: No parameterless constructor defined" when calling BulkInsertAsync

### ISSUE THREAD

**Issue by crperez5** [borisdj/EFCore.BulkExtensions#755]
Hi there, 

we are getting the exception described on the title whenever we call BulkInsertAsync passing a list of objects that do not have a parameterless constructor. The same code used to work just fine in version 3.6.3. We started having this issue from version 6.3.9 (we migrated to .net 6).

The code:

`         await _context.BulkInsertAsync(messages.ToList(), config =>
         {
            config.SetOutputIdentity = true;
            config.BatchSize = 2000;
            configFunc?.Invoke(config);
         });`

**Comment by CDSGlobalUK** [comment]
I looked at the source code, and it relates to line below (GenericsHelper.cs), I rolled back, but I also tried it with that line commented out and it worked again

`  if (!type.IsInterface)
                    **defaultValue = field.GetValue(Activator.CreateInstance(type, true));**
                
                if (temp == defaultValue)
                    result.Add(name);

                if (temp is Guid guid && guid == Guid.Empty)
                    result.Add(name);`

**Comment by borisdj** [comment]
Still could you check and confirm if this would work as a fix:
```
var hasParameterlessConstructor = type.GetConstructor(Type.EmptyTypes) != null;
if (!type.IsInterface && hasParameterlessConstructor)
   defaultValue = field.GetValue(Activator.CreateInstance(type, true));
```

**Comment by CDSGlobalUK** [comment]
@borisdj The above works for me, thanks

**Comment by borisdj** [comment]
Great.
Btw I have tried to make a test for it but get exception when getting DbSet of context or Database:.
`No suitable constructor was found for entity type ''. The following constructors had parameters that could not be bound to properties of the entity type`
Do you have example of Entity defined in context.

**Comment by cyril265** [comment]
> Does EF itself requires parameterless constructor: [SO.par.const.](https://stackoverflow.com/questions/31543255/why-must-i-have-a-parameterless-constructor-for-code-first-entity-framework/31543487) If that is the case then how can you have entity for Bulk without it. Also not sure how to write a test since for it.

The answer is outdated. Current version of EF doesn't require parameterless constructors https://docs.microsoft.com/en-us/ef/core/modeling/constructors

**Comment by borisdj** [comment]
Published v 6.4.1
